src/controllers/institution-controller.js

```diff
@@ -68,6 +68,91 @@ export async function addInstitutionToProject(req, res) {
   }
 }
 
+export async function editInstitution(req, res) {
+  const projectId = req.params.projectId
+  const institutionId = req.body.institutionId
+  const name = req.body.name
+
+  const institution = await models.Institution.findByPk(institutionId)
+
+  let presentInstitutionWithName = await models.Institution.findOne({
+    where: { name: name },
+  })
+
+  const dupeProjectInstitutionIds =
+    await institutionService.getInstitutionProjectReferences(projectId, [
+      institutionId,
+    ])
+  const institutionxUserIds =
+    await institutionService.getInstitutionUserReferences([institutionId])
+
+  const uniqueInstitutionIds =
+    dupeProjectInstitutionIds.length || institutionxUserIds.length
+      ? []
+      : [institutionId]
```

Comment:
Suggestion: `uniqueInstitutionIds` is not too intuitive. Let's change:
`const institutionInUse = dupeProjectInstitutionIds.length || institutionxUserIds.length`

and then `uniqueInstitutionIds.length == 1` becomes `!institutionInUse`

and then the below condition becomes:
```
if (!institutionInUse) {
    await models.Institution.destroy({
      where: {
        institution_id: institutionId,
      },
      transaction: transaction,
      individualHooks: true,
      user: req.user,
    })
}
```

This is much easier to read the code and understand what it does.